ale_linters/rust/cargo.vim

```diff
@@ -3,19 +3,89 @@
 
 
 function! ale_linters#rust#cargo#GetCargoExecutable(bufnr) abort
-    if ale#util#FindNearestFile(a:bufnr, 'Cargo.toml') !=# ''
-        return 'cargo'
-    else
-        " if there is no Cargo.toml file, we don't use cargo even if it exists,
-        " so we return '', because executable('') apparently always fails
-        return ''
-    endif
+	if ale#util#FindNearestFile(a:bufnr, 'Cargo.toml') !=# ''
+		return 'cargo'
+	else
+		" if there is no Cargo.toml file, we don't use cargo even if it exists,
+		" so we return '', because executable('') apparently always fails
+		return ''
+	endif
+endfunction
+
+function! ale_linters#rust#cargo#HandleCargoErrors(buffer_number, errorlines) abort
```

Comment:
Here is output from `cargo rustc --lib -- --error-format=json -Z no-trans`:
```
{"message":"expected one of `!` or `::`, found `hyper`","code":null,"level":"error","spans":[{"file_name":"src/crawl/mod.rs","byte_start":1529,"byte_end":1534,"line_start":7,"line_end":7,"column_start":4,"column_end":9,"is_primary":true,"text":[{"text":"se hyper::Url;","highlight_start":4,"highlight_end":9}],"label":null,"suggested_replacement":null,"expansion":null}],"children":[],"rendered":null}
```

And here is the output from `cargo build --message-format json`:
```
{"message":{"children":[],"code":null,"level":"error","message":"expected one of `!` or `::`, found `hyper`","rendered":null,"spans":[{"byte_end":1534,"byte_start":1529,"column_end":9,"column_start":4,"expansion":null,"file_name":"src/crawl/mod.rs","is_primary":true,"label":null,"line_end":7,"line_start":7,"suggested_replacement":null,"text":[{"highlight_end":9,"highlight_start":4,"text":"se hyper::Url;"}]}]},"package_id":"unibot 0.1.0 (path+file:///home/notkild/projects/rust/unibot)","reason":"compiler-message","target":{"kind":["lib"],"name":"libunibot","src_path":"/home/notkild/projects/rust/unibot/src/lib.rs"}}
```